crates/core/src/host/wasm_common/module_host_actor.rs

```diff
@@ -708,6 +707,63 @@ impl<T: WasmInstance> WasmModuleInstance<T> {
         let inner = self.database_instance_context().logger.lock().unwrap();
         SystemLogger { inner }
     }
+
+    /// Compute the diff between the current and proposed schema.
+    fn schema_updates(&self, tx: &MutTxId) -> anyhow::Result<SchemaUpdates> {
+        let stdb = &*self.database_instance_context().relational_db;
+
+        let mut new_tables = HashMap::new();
+        let mut tainted_tables = Vec::new();
+
+        let mut known_tables: BTreeMap<String, TableSchema> = stdb
+            .get_all_tables(tx)?
+            .into_iter()
+            .map(|schema| (schema.table_name.clone(), schema))
+            .collect();
+
+        for table in self.info.catalog.values().filter_map(EntityDef::as_table) {
+            let mut proposed_schema = self.schema_for(table)?;
+            if let Some(known_schema) = known_tables.remove(&table.name) {
+                // If the table is known, we also know its id. Update the
+                // index definitions so the `TableDef` of both schemas is
+                // equivalent.
+                for index in proposed_schema.indexes.iter_mut() {
+                    index.table_id = known_schema.table_id;
+                }
+                let known_schema = TableDef::from(known_schema);
+                if known_schema != proposed_schema {
+                    self.system_logger()
+                        .warn(&format!("stored and proposed schema of `{}` differ", table.name));
+                    tainted_tables.push(table.name.to_owned());
+                } else {
+                    // Table unchanged
+                }
```

Comment:
```suggestion
                // If the schema has changed, the table is tainted.
                // If the schemas match, everything is peachy and we can continue.
                if known_schema != proposed_schema {
                    self.system_logger()
                        .warn(&format!("stored and proposed schema of `{}` differ", table.name));
                    tainted_tables.push(table.name.to_owned());
                }
```

Empty `else` block is a code smell to me. I'd rather see a comment introducing the conditional itself.

Reply:
c89b358816c116a729f4a0c1096c3ed581cb68ca